### PR TITLE
docs(log) http-log authentication and retries logging

### PR DESCRIPTION
**Pull request by Tieske**
Deals with Kong issue https://github.com/Mashape/kong/pull/2429: logging retries and failure info
and https://github.com/Mashape/kong/pull/2430 for adding a basic auth header with the http log plugin

This documents the additional changes in https://github.com/Mashape/kong/pull/2467.

### CODE REVIEW

File: app/plugins/file-log.md [hunk 1]
```diff
@@ -77,9 +77,21 @@ Every request will be logged separately in a JSON object separated by a new line
             "access-control-allow-origin": "*"
         }
     },
+    "tries": [
```
Comment (p0pr0ck5) [inline app/plugins/file-log.md]: this is valid for the file-log plugin?

File: app/plugins/http-log.md [hunk 2]
```diff
@@ -40,11 +40,15 @@ form parameter                  | default | description
 ---                             | ---     | ---
 `name`                          |         | The name of the plugin to use, in this case: `http-log`
 `consumer_id`<br>*optional*     |         | The CONSUMER ID that this plugin configuration will target. This value can only be used if [authentication has been enabled][faq-authentication] so that the system can identify the user making the request.
-`config.http_endpoint`          |         | The HTTP endpoint (including the protocol to use) where to send the data to
+`config.http_endpoint`          |         | The HTTP endpoint (including the protocol to use) where to send the data to.
```
Comment (theetrain) [inline app/plugins/http-log.md]: > The HTTP endpoint (including the protocol) `to which the data will be sent.`

File: app/plugins/file-log.md (hunk 1 above)
Comment (thibaultcha) [inline app/plugins/file-log.md]: Curious: why not? It should use the same serializer as the other ones, so yes. Maybe your understanding of the file-log plugin is that it doesn't log requests/response info?

File: app/plugins/http-log.md [hunk 3]
```diff
@@ -40,11 +40,15 @@ form parameter                  | default | description
...
 `config.method`<br>*optional*   | `POST`  | An optional method used to send data to the http server, other supported values are PUT, PATCH
 `config.timeout`<br>*optional*  | `10000` | An optional timeout in milliseconds when sending data to the upstream server
 `config.keepalive`<br>*optional*| `60000` | An optional value in milliseconds that defines for how long an idle connection will live before being closed
 
+**NOTE:** If the `config.http_endpoint` contains a username and password, then
```
Comment (thibaultcha) [inline app/plugins/http-log.md]: I think we should make clear "where":
```
contains a username and password in the authentication section of the URL.
Ex: http://bob:password@example.com/logs
```

File: app/plugins/file-log.md [hunk 4]
```diff
@@ -112,6 +124,7 @@ A few considerations on the above JSON object:
 
 * `request` contains properties about the request sent by the client
 * `response` contains properties about the response sent to the client
+* `tries` contains the information of the successive (re)tries, the upstream address and failure information
```
Comment (thibaultcha) [inline app/plugins/file-log.md]: I think a bit more context here wouldn't hurt:
```
contains the list of (re)tries (successes and failures) made by the load balancer for this request
```

File: app/plugins/file-log.md (hunk 1 above)
Comment (p0pr0ck5) [inline app/plugins/file-log.md]: Yep, misread this. Thought this had to do with the log plugin retrying, not the retries in the serialized data. My bad!

**Pull request merged**